analyses/transcriptomic-dimension-reduction/README.md

```diff
@@ -0,0 +1,130 @@
+# Dimension reduction: gene expression data
+
+This module runs dimension reduction—namely PCA, UMAP, and t-SNE—on gene expression data from kallisto and RSEM.
+
+To the run the whole pipeline and produce 4 plots with panels for all three approaches
+
+1. RSEM stranded data
+2. RSEM poly-A data
+3. kallisto stranded data
+4. kallisto poly-A data
+
+with points colored by broad histology, simply run:
+
+```
+./dimension-reduction-plots.sh
+```
+
+## Notes on further applications
+
+To document this module beyond what is in the individual files, we list some example applications or extensions below.
+
+Running dimension reduction, creating lists of plots, and generating the multipanel plots can all be run separately using the appropriate Rscripts in `scripts/`:
+
+```
+generate-multipanel-plot.R
+get-plot-list.R
+run-dimension-reduction.R
+```
+
+### Repeat dimension reduction step with different seeds
+
+##### Command line
+
+If we wanted to run UMAP multiple times with different seed values on the kallisto poly-A data, but _skip t-SNE_, we could do the following:
+
+```sh
+seeds=(309 4642 2019 2994 5783)
+
+for seed in "${seeds[@]}"
+do
+  Rscript --vanilla scripts/run-dimension-reduction.R \
+    --expression ../../data/pbta-gene-expression-kallisto.polya.rds \
+    --metadata ../../data/pbta-histologies.tsv \
+    --filename_lead kallisto_polyA_${seed} \
+    --output_directory results \
+    --seed ${seed} \
+    --skip_tsne
+done
+
+```
+
+Unfortunately, this approach will also run PCA 5 times. 
+
+##### R
+
+It's also possible to repeat this UMAP step 5 times within R:
+
+```R
+# load the required custom functions
+source("util/dimension-reduction-functions.R")
+
+# read in kallisto poly-A data and transpose
+expression_file <- "../../data/pbta-gene-expression-kallisto.polya.rds"
+
+# Read in expression data
```

Comment:
Sometimes you capitalize your comments and sometimes you don't. This doesn't matter but I'm just stating this because it's something my brain notices.

Reply:
I think technically @cbethell capitalizes comments and I don't 😁